olmo/scaling/scaling_laws/extrapolate_n.py

```diff
@@ -0,0 +1,70 @@
+import csv
+from dataclasses import dataclass
+from typing import List, Dict
+
+import matplotlib.pyplot as plt
+import numpy as np
+import scipy
+
+from .curve_fit import get_coefficients
+
+
+@dataclass
+class CurveFitConfig:
```

Comment:
@liujch1998 We don't need 3 different versions of the CurveFitConfig, do we? It can all be rolled into one

Reply:
The config needed for running extrapolation on N is quite different from that of extrapolating D (i.e. curve_fit.py), so I thought it'd be nice to keep them separate so it's clear what is required. I can rename them into distinct class names.